back/src/Model/GameRoom.ts

```diff
@@ -249,32 +248,87 @@ export class GameRoom {
+                    for (const member of followingMembers) {
+                        const distance = GameRoom.computeDistanceBetweenPositions(
+                            member.getPosition(),
+                            previewNewGroupPosition
+                        );
+
+                        if (distance > this.groupRadius) {
+                            isOutOfBounds = true;
+                            break;
+                        }
+                    }
+                    group.setOutOfBounds(isOutOfBounds);
                 }
-                const usrPos = foreignUser.getPosition();
-                const grpPos = foreignUser.group.getPosition();
-                const distance = GameRoom.computeDistanceBetweenPositions(usrPos, grpPos);
+            }
+
+            // Check if the moving user has kicked out another user
+            for (const headMember of group.getGroupHeads()) {
+                if (!headMember.group) {
+                    this.leaveGroup(headMember);
+                    continue;
+                }
+
+                const headPosition = headMember.getPosition();
+                const distance = GameRoom.computeDistanceBetweenPositions(headPosition, previewNewGroupPosition);
 
                 if (distance > this.groupRadius) {
-                    if (foreignUser.hasFollowers() || foreignUser.following) {
-                        // If one user is out of the group bounds BUT following, the group still exists... but should be hidden.
-                        // We put it in 'outOfBounds' mode
-                        group.setOutOfBounds(true);
-                        noOneOutOfBounds = false;
-                    } else {
-                        this.leaveGroup(foreignUser);
+                    hasKickOutSomeone = true;
+                    break;
+                }
+            }
+
+            /**
+             * If the current moving user has kicked another user from the radius,
+             * the moving user leaves the group because he is too far away.
+             */
+            const userDistance = GameRoom.computeDistanceBetweenPositions(user.getPosition(), previewNewGroupPosition);
+
+            if (hasKickOutSomeone && userDistance > this.groupRadius) {
+                this.leaveGroup(user);
+
+                if (user.hasFollowers()) {
+                    for (const member of followingMembers) {
+                        this.leaveGroup(member);
                     }
+
+                    // Re-create a group with the followers
+                    const newGroup: Group = new Group(
+                        this.roomUrl,
+                        [user, ...followingMembers],
+                        this.groupRadius,
+                        this.connectCallback,
+                        this.disconnectCallback,
+                        this.positionNotifier
+                    );
+                    this.groups.add(newGroup);
                 }
```

Comment:
Not perfect. If we have 3 users (1 leader, 1 follower and 1 not following):

If the leader moves away, the group is destroyed AND recreated for the leader and the follower.

Because the group is destroyed and recreated:

- We lose screen sharing
- We get a ding-dong sound

If the result of the split operation is that only one user will leave the group, then the group should not be destroyed.